Amplify/Core/Support/Internal/NSLocking+Execute.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// - Warning: Although this has `public` access, it is intended for internal use
+/// and should not be used directly by host applications. The behaviors and names of
+/// this type may change without warning.
+extension NSLocking {
+    /// Execute `block` after obtaining a lock on `lock`.
+    ///
+    /// - Warning: Although this has `public` access, it is intended for internal use
```

Comment:
do you have to make it public for some reason and thus need this warning?

Reply:
Yes, in order to use Amplify methods in the separate packages, we have to have public accessibility.

Reply:
ah gotcha makes sense, I understand this well